File: paper_artifacts/experiments/llr8/artifacts/sources/oss120b-fortran/tsvc_2_s152__submitted.f90
subroutine tsvc_2_s152_fp64(a, b, c, d, e, len_1d, workspace, workspace_bytes) bind(C, name="tsvc_2_s152_fp64")
    use iso_c_binding
    implicit none
    integer(c_int64_t), value :: len_1d
    type(c_ptr), value :: workspace
    integer(c_int64_t), value :: workspace_bytes
    real(c_double), intent(inout) :: a(*)
    real(c_double), intent(inout) :: b(*)
    real(c_double), intent(in) :: c(*)
    real(c_double), intent(in) :: d(*)
    real(c_double), intent(in) :: e(*)
    integer :: i
    real(c_double) :: tmp
    !$omp parallel do default(none) shared(a,b,c,d,e,len_1d) private(i,tmp)
    do i = 1, len_1d
        tmp = d(i) * e(i)
        b(i) = tmp
        a(i) = a(i) + tmp * c(i)
    end do
    !$omp end parallel do
end subroutine tsvc_2_s152_fp64

File: paper_artifacts/experiments/llr9/artifacts/sources/qwen38-c/tsvc_2_s152__submitted.c
#include <stdint.h>

void tsvc_2_s152_fp64(double* a, double* b, double* c, double* d, double* e,
                      int64_t LEN_1D, uint8_t* ws, int64_t ws_bytes)
{
    (void)ws; (void)ws_bytes;
    #pragma omp parallel for schedule(static)
    for (int64_t i = 0; i < LEN_1D; i++) {
        double t = d[i] * e[i];
        b[i] = t;
        a[i] = a[i] + t * c[i];
    }
}
